app/discovery/provider/consulcatalog/list.go

```diff
@@ -0,0 +1,80 @@
+package consulcatalog
+
+import (
+	"fmt"
+	log "github.com/go-pkgz/lgr"
+	"github.com/umputun/reproxy/app/discovery"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+func (cc *ConsulCatalog) List() ([]discovery.URLMapper, error) {
+	log.Print("[DEBUG] call consul catalog list")
+
+	cc.listMx.RLock()
+	defer cc.listMx.RUnlock()
```

Comment:
I don't get why this needs to be locked. Maybe there is a reason, but I don't see it in this file

Reply:
ConsulCatalog (CC) saves services info in the internal ConsulCatalog.list field. When CC send the update signal to the events channel, CC dont know - when exactly Reproxy will called List method. For this reason I use mutex